### ISSUE TITLE
💅 `useTemplate` doesn't report expressions with an interpolated template literal and non-string expressions

### ISSUE THREAD

**Issue by n-gude** [biomejs/biome#1456]
### Environment information

```bash
N/A
```

### Rule name

useTemplate

### Playground link

https://biomejs.dev/playground/?code=LwAvACAARABvAGUAcwBuACcAdAAgAHIAZQBwAG8AcgB0ADoACgBgAGEAJAB7ADEAfQBgACAAKwAgADIAOwAKAAoALwAvACAAVwBoAGkAbABlACAAcgBlAHAAbwByAHQAaQBuAGcAOgAKACIAYQAiACAAKwAgADEAOwAKAGAAYQBgACAAKwAgADEAOwAKAGAAYQAkAHsAMQB9AGAAIAArACAAIgBiACIAOwAKAA%3D%3D

### Expected result

The following code should be reported:

```js
`a${1}` + 2;
```

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by n-gude** [comment]
This will be closed by #1206.
